### PR TITLE
Add `setjmp` and `longjmp` functions for stm32f4-nuttx target

### PR DESCRIPTION
**Pull request by robertsipka**

### CODE REVIEW

File: config/nuttx/stm32f4dis/app/setjmp.S [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
```
Comment (yichoi) [inline config/nuttx/stm32f4dis/app/setjmp.S]: Does this file come from JerryScript ? I wonder whether we should change the license clauses or not.

**Comment by glistening** [comment]
@robertsipka Could you share what jerry tests are passed by this change?

**Comment by robertsipka** [comment]
The setjmp implementation fixes the `syntax_fail1.js` test case on stm32f4-nuttx target.

Added the `setjmp` and `longjmp` functions on jerry side fixes several tests:

```
Jerry common tests with setjmp (after):
    297 PASS
     10 FAIL 

Jerry common tests without setjmp (before):
    245 PASS
     13 FAIL
     49 TIMEOUT (CRASH)

```

Note: This measure also contains [this](https://github.com/jerryscript-project/jerryscript/pull/1744) PR, so these changes together give the results.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**